### PR TITLE
Issue 107: Make zstd compressionLevel configurable

### PR DESCRIPTION
**Pull request by shangxinli**

### CODE REVIEW

File: src/main/java/io/airlift/compress/zstd/ZstdCompressor.java [hunk 1]
```diff
@@ -97,9 +99,14 @@ else if (output.hasArray()) {
...
                         outputAddress,
                         outputLimit,
-                        CompressionParameters.DEFAULT_COMPRESSION_LEVEL);
+                        compressionLevel);
                 output.position(output.position() + written);
             }
         }
     }
+
+    public void setCompressionLevel(int compressionLevel)
```
Comment (electrum) [inline src/main/java/io/airlift/compress/zstd/ZstdCompressor.java:108]: I don't think we should make this class have mutable state. We could make the field `final` make it a constructor argument. We'd want to preserve the existing no-argument constructor for simplicity and compatibility.

@martint thoughts?

Comment (DinoCassowary) [reply src/main/java/io/airlift/compress/zstd/ZstdCompressor.java:108]: Why not have an overload of `compress` that accepts this input? I feel like I'm missing something.

Comment (jackluo923) [reply src/main/java/io/airlift/compress/zstd/ZstdCompressor.java:108]: @DinoCassowary  Only compression level 3 and 4 via `Strategy.DFast` are supported. Not sure if setting the compression level is even necessary at the moment.

**Comment by dain** [comment]
Only the default level is supported (implemented) right now, so it doesn’t make sense to allow it to be configurable at this time

**Pull request closed without merging**